### PR TITLE
Align parser with new swan lake service changes

### LINKED ISSUE #25797: Align parser with new swan lake service changes
**Description:**
$subject.

**Suggested Labels:**
Team/CompilerFE, Area/Parser, Type/NewFeature

**Suggested Assignees:**
@lochana-chathura

### PR DESCRIPTION
**Pull request by lochana-chathura**
## Purpose

- Remove `service-constructor-expr` support
- Remove  `service-type-descriptor` support
- Add `service` qualifier support to `module-class-defn`, `object-type-descriptor` and `object-constructor-expr`
- Add `resource` qualifier support to `object-field`
- Add  new `service-decl` support. 
- Add new `resource-accessor-defn` support to `module-class-defn` and `object-contructor-expr`
- Replace `resource` attach point with `service remote`
- Disallow `visibility-qual` with `remote-qual` and `resource-qual`
</details>
<details>
  <summary><b>service-decl</b> (<i>click to expand</i>)</summary>

```
service-decl := metadata service [type-descriptor] [absolute-resource-path] on expression-list object-constructor-block
absolute-resource-path := "/" | ("/" identifier)+
```

</details>

</details>
<details>
  <summary><b>resource-accessor-defn</b> (<i>click to expand</i>)</summary>

```
resource-accessor-defn := metadata `resource` `function` accessor-name relative-resource-path
                           function-signature method-defn-body
accessor-name := identifier
relative-resource-path := "." | (identifier ("/" identifier)*)
```

</details>

Fixes #25797

## Approach
N/A

## Samples
N/A

## Remarks
N/A

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [x] Added necessary documentation  
  - [x] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java [hunk 1]
```diff
@@ -1031,125 +1028,131 @@ public BLangNode transform(ObjectFieldNode objFieldNode) {
...
-            BLangServiceConstructorExpr serviceConstructorExpr =
-                    (BLangServiceConstructorExpr) TreeBuilder.createServiceConstructorNode();
-            serviceConstructorExpr.serviceNode = bLService;
-            return serviceConstructorExpr;
-        }
+        simpleVar.pos = getPositionWithoutMetadata(objFieldNode);
+        return simpleVar;
     }
 
+//    @Override
```
Comment (rdhananjaya) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:1042]: Shall we remove this commented code

Comment (lochana-chathura) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:1042]: sure

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java [hunk 2]
```diff
@@ -1031,125 +1028,131 @@ public BLangNode transform(ObjectFieldNode objFieldNode) {
...
+//            return var;
+//        } else {
+//            BLangServiceConstructorExpr serviceConstructorExpr =
+//                    (BLangServiceConstructorExpr) TreeBuilder.createServiceConstructorNode();
+//            serviceConstructorExpr.serviceNode = bLService;
+//            return serviceConstructorExpr;
+//        }
+//    }
+
     public void addServiceConstructsToClassDefinition(ServiceBodyNode serviceBodyNode,
```
Comment (rdhananjaya) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:1042]: this looks like dead code, if so shall we remove it

Comment (lochana-chathura) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:1042]: yea. will remove. thanks

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/diagnostics/DiagnosticErrorCode.java [hunk 3]
```diff
@@ -241,17 +241,12 @@
...
     // Expression followed by the start keyword must be a func-call, a method-call or a remote-method-call
     ERROR_INVALID_EXPRESSION_IN_START_ACTION("BCE207", "error.invalid.expression.in.start.action"),
-    // Cannot have the  same qualifier twice
-    ERROR_DUPLICATE_OBJECT_TYPE_QUALIFIER("BCE208", "error.duplicate.object.type.qualifier"),
-    ERROR_DUPLICATE_CLASS_TYPE_QUALIFIER("BCE209", "error.duplicate.class.type.qualifier"),
-    ERROR_DUPLICATE_OBJECT_METHOD_QUALIFIER("BCE210", "error.duplicate.object.method.qualifier"),
-    ERROR_QUALIFIER_NOT_ALLOWED("BCE211", "error.qualifier.not.allowed"),
-    ERROR_ABSTRACT_QUALIFIER_NOT_ALLOWED_IN_OBJECT("BCE212", "error.abstract.qualifier.not.allowed.in.object"),
-    ERROR_READONLY_QUALIFIER_NOT_ALLOWED_IN_OBJECT("BCE213", "error.readonly.qualifier.not.allowed.in.object"),
+    // Cannot have the same qualifier twice
```
Comment (rdhananjaya) [inline compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/diagnostics/DiagnosticErrorCode.java:244]: Please check if all messages removed here are removed from *message.properties files

**Review by rdhananjaya — APPROVED** [review]

**Pull request merged**